lib/memo_wise.rb

```diff
@@ -94,42 +183,164 @@ def #{method_name}#{args_str}
     end
   end
 
-  def reset_memo_wise(method_name, *args, **kwargs)
+  # Presets the memoized result for the given method to the result of the given
+  # block.
+  #
+  # This method is for situations where the caller *already* has the result of
+  # an expensive method call, and wants to preset that result as memoized for
+  # future calls. In other words, the memoized method will be called *zero*
+  # times rather than once.
```

Comment:
Do we want to explicitly name in the docs that we are not validating parameters? That is to say, if a set of invalid parameters is passed to this method, we will *not* raise an error

Reply:
@jemmaissroff Sure, I can add that if you'd like

Reply:
Ok, added @jemmaissroff, please review and resolve this thread if it's good?